qcodes/dataset/plotting.py

```diff
@@ -269,6 +272,110 @@ def plot_by_id(run_id: int,
     return axes, new_colorbars
 
 
+def _complex_to_real_preparser(alldata: NamedData,
+                               conversion: str,
+                               degrees: bool=False) -> NamedData:
+    """
+    Convert complex-valued parameters to two real-valued parameters, either
+    real and imaginary part or phase and magnitude part
+
+    Args:
+        alldata: the data to convert, should be the output of get_data_by_id
+        conversion: the conversion method, either "real_and_imag" or
+            "phase_and_mag"
+        degress: whether to return the phase in degrees. The default is to
+            return the phase in radians
+    """
+
+    if conversion not in ['real_and_imag', 'phase_and_mag']:
+        raise ValueError(f'Invalid conversion given. Received {conversion}, '
+                         'but can only accept "real_and_imag" or '
+                         '"phase_and_mag".')
```

Comment:
awesome!

`mag_and_phase`? :)

Reply:
Hmm, that's probably true. `(r, theta)` sounds natural.